src/robot/conf/settings.py

```diff
@@ -157,6 +159,13 @@ def _validate_log_level_and_default(self, log_level, default):
             raise DataError("Default visible log level '%s' is lower than "
                             "log level '%s'" % (default, log_level))
 
+    def _process_max_error_lines(self, name, value):
+        value = self._convert_to_integer(name, value)
+        if value < 0:
+            raise DataError("Option '--%s' expected positive integer value but got '%s'."
+                            % (name.lower(), value))
+        return value
```

Comment:
1. We could consider making it so that negative values like `-1` means not cutting the message at all. Alternatively could use `NONE` for that purpose because `--maxerrorlines NONE` would be more explicit. Not sure is this really needed, though.

2. Very small values, especially 0, aren't really valid. They'd make the message *really* short and I'm not sure would our splitting algorithm work with such values either. Should we make it so that the value should be at least 10?

Reply:
1. Opted for using `NONE` for not cutting the message. `-1` seems a bit confusing to me and would require an extra check in `_process_max_error_lines()` from `settings.py`. Documented this in `CreatingTestCases.rst` - not sure if it has to be specified in other places too.
2. A limit of 10 sounds good.